src/scoping-shim.js

```diff
@@ -88,29 +88,31 @@ export default class ScopingShim {
     template.extends = typeExtension;
     templateMap[elementName] = template;
     let cssBuild = StyleUtil.getCssBuild(template);
-    let cssText = this._gatherStyles(template);
+    const optimalBuild = StyleUtil.isOptimalCssBuild(cssBuild);
     let info = {
       is: elementName,
       extends: typeExtension,
     };
+    let cssText = this._gatherStyles(template);
     // check if the styling has mixin definitions or uses
     this._ensure();
-    let hasMixins = !StyleUtil.elementHasBuiltCss(template) && detectMixin(cssText);
-    let ast = parse(cssText);
-    // only run the applyshim transforms if there is a mixin involved
-    if (hasMixins && nativeCssVariables && this._applyShim) {
-      this._applyShim['transformRules'](ast, elementName);
+    if (!optimalBuild) {
+      let hasMixins = !StyleUtil.elementHasBuiltCss(template) && detectMixin(cssText);
```

Comment:
Can you explain how `elementHasBuildCss` returns different info from what's in `cssBuild`?

Reply:
Ah, that can just be `cssBuild`, as `elementHasBuildCss` is effectively `cssBuild !== ''`